Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/QuadrigaRoleManager.java

```diff
@@ -124,38 +120,49 @@ public void fillQuadrigaRole(String type, IQuadrigaRole collaboratorRole) {
         List<IQuadrigaRole> roles = rolesMap.get(type);
         if (roles == null)
             return;
-        
-        for(IQuadrigaRole role: roles)
-        {
-            if(role.getDBid().equals(collaboratorRole.getDBid()))
-            {
+
+        for (IQuadrigaRole role : roles) {
+            if (role.getDBid().equals(collaboratorRole.getDBid())) {
                 collaboratorRole.setId(role.getId());
                 collaboratorRole.setDBid(role.getDBid());
                 collaboratorRole.setName(role.getName());
                 collaboratorRole.setDescription(role.getDescription());
                 collaboratorRole.setDisplayName(role.getDisplayName());
             }
-        } 
-    } 
-
-	/**
-	 * {@inheritDoc}
-	 */
-	@Override
-	public List<IQuadrigaRole> getQuadrigaRoles(String type) {
-	    List<IQuadrigaRole> listForType = rolesMap.get(type);
-	    List<IQuadrigaRole> returnList = new ArrayList<IQuadrigaRole>(listForType);
-		return returnList;
-	}
-	
-	@Override
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<IQuadrigaRole> getQuadrigaRoles(String type) {
+        List<IQuadrigaRole> listForType = rolesMap.get(type);
+        List<IQuadrigaRole> returnList = new ArrayList<IQuadrigaRole>(listForType);
+        return returnList;
+    }
+
+    @Override
     public List<IQuadrigaRole> getSelectableQuarigaRoles(String type) {
-	    List<IQuadrigaRole> selectableRoles = new ArrayList<IQuadrigaRole>();
-	    for (IQuadrigaRole role : rolesMap.get(type)) {
-	        if (role.isSelectable())
-	            selectableRoles.add(role);
-	    }
-	    return selectableRoles;
-	}
-
-}
+        List<IQuadrigaRole> selectableRoles = new ArrayList<IQuadrigaRole>();
+        for (IQuadrigaRole role : rolesMap.get(type)) {
+            if (role.isSelectable())
+                selectableRoles.add(role);
+        }
+        return selectableRoles;
+    }
+
+    @Override
+    public List<IQuadrigaRole> getProjectCollaboratorRoles(String type) {
```

Comment:
This doesn't make much sense. You parameterize the method, but its name and the logic is specifically for one type. What would be the point of the parameter? Just get all role names and remove the owner role in your controller. It's not the prettiest solution but since you only need it in one place, let's go with the easy fix for now.